tests/integration/test_kfp_functional_v2.py

```diff
@@ -0,0 +1,229 @@
+    )
+    pipeline_version_id = (
+        kfp_client.list_pipeline_versions(pipeline_upload_response.pipeline_id)
+        .pipeline_versions[0]
+        .pipeline_version_id
+    )
+
+    yield pipeline_upload_response, pipeline_version_id
+
+    kfp_client.delete_pipeline_version(pipeline_upload_response.pipeline_id, pipeline_version_id)
+    kfp_client.delete_pipeline(pipeline_upload_response.pipeline_id)
+
+
+@pytest.fixture(scope="function")
+def create_and_clean_experiment_v2(kfp_client: kfp.Client):
+    """Create an experiment and remove after test case execution."""
+    experiment_response = kfp_client.create_experiment(
+        name="test-experiment", namespace=KUBEFLOW_PROFILE_NAMESPACE
+    )
+
+    yield experiment_response
+
+    kfp_client.delete_experiment(experiment_id=experiment_response.experiment_id)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest, request, lightkube_client):
+    """Build and deploy kfp-operators charms."""
+    no_build = request.config.getoption("no_build")
+
+    # Immediately raise an error if the model name is not kubeflow
+    if ops_test.model_name != "kubeflow":
+        raise ValueError("kfp must be deployed to namespace kubeflow")
+
+    # Get/load template bundle from command line args
+    bundlefile_path = Path(request.config.getoption("bundle"))
+    basedir = Path("./").absolute()
+
+    # Build the charms we need to build only if --no-build is not set
+    context = {}
+    if not no_build:
+        charms_to_build = {
+            charm: Path(CHARM_PATH_TEMPLATE.format(basedir=str(basedir), charm=charm))
+            for charm in KFP_CHARMS
+        }
+        log.info(f"Building charms for: {charms_to_build}")
+        built_charms = await ops_test.build_charms(*charms_to_build.values())
+        log.info(f"Built charms: {built_charms}")
+
+        for charm, charm_file in built_charms.items():
+            charm_resources = get_resources_from_charm_file(charm_file)
+            context.update([(f"{charm.replace('-', '_')}_resources", charm_resources)])
+            context.update([(f"{charm.replace('-', '_')}", charm_file)])
+
+    # Render kfp-operators bundle file with locally built charms and their resources
+    rendered_bundle = render_bundle(
+        ops_test, bundle_path=bundlefile_path, context=context, no_build=no_build
+    )
+
+    # Deploy the kfp-operators bundle from the rendered bundle file
+    await deploy_bundle(ops_test, bundle_path=rendered_bundle, trust=True)
+
+    # Wait for everything to be up.  Note, at time of writing these charms would naturally go
+    # into blocked during deploy while waiting for each other to satisfy relations, so we don't
+    # raise_on_blocked.
+    await ops_test.model.wait_for_idle(
+        status="active",
+        raise_on_blocked=False,  # These apps block while waiting for each other to deploy/relate
+        raise_on_error=True,
+        timeout=3600,
+        idle_period=30,
```

Comment:
Here we have 30 seconds. WRT my other comment, if there isn't a reason to have different idle_periods when using sdk_v1 and sdk_v2, we should decide and keep those the same in both places.

Reply:
I don't think this was intentional, maybe a rebase error. I think I can leave it on 30s, let me just change it to 30s based also in your other comment.